### PR TITLE
fix(qc): allow specifying severity in lower case

### PR DESCRIPTION
**Pull request by kulgan**

### CODE REVIEW

File: sheepdog/blueprint/qcreport/service.py [hunk 1]
```diff
@@ -31,17 +30,28 @@ def from_value(cls, key):
...
+        if self.value == "severity" and qcreport.SEVERITY.enums.count(value.upper()) == 0:
+            raise ValueError("{} not allowed for {}, allowed values include {}".
+                             format(value, "severity", qcreport.SEVERITY.enums))
+        return value.upper() if self.value == "severity" else value
+
 
 class SummaryFilter(ValidationResultFilter):
     NODE_TYPE = "node_type"
     SEVERITY = "severity"
+    ERROR_TYPE = "error_type"
```
Comment (anmaxvl) [inline sheepdog/blueprint/qcreport/service.py]: 🤔 I guess it's fine...

File: sheepdog/blueprint/qcreport/service.py [hunk 2]
```diff
@@ -156,22 +167,22 @@ def list_reports(self, filters=None, sort_by=None, offset=None, limit=100):
                     result = entry.ValidationResult
                     if result.error_type not in summary:
                         summary[result.error_type] = {"description": result.message, "severity": result.severity,
-                                                     "count": 0}
+                                                      "count": 0}
                     summary[result.error_type]["count"] += 1
                     response["items"].append(result.to_json())
             except DataError as e:
-                raise InvalidDataError(e.message)
+                raise ValueError(e.message)
```
Comment (anmaxvl) [inline sheepdog/blueprint/qcreport/service.py]: `str(e)`

File: sheepdog/blueprint/qcreport/service.py [hunk 3]
```diff
@@ -31,17 +30,28 @@ def from_value(cls, key):
...
 
+    def validate(self, value):
+        """ Validates to ensure value matches database expectations
+        Args:
+            value (str):
+        """
+        if self.value == "severity" and qcreport.SEVERITY.enums.count(value.upper()) == 0:
+            raise ValueError("{} not allowed for {}, allowed values include {}".
+                             format(value, "severity", qcreport.SEVERITY.enums))
+        return value.upper() if self.value == "severity" else value
```
Comment (scintillating7) [inline sheepdog/blueprint/qcreport/service.py]: for future reference, looks like you can define a sqlalchemy enum from a python enum, which I would hope would make comparisons like this less gross https://docs.sqlalchemy.org/en/13/core/type_basics.html#sqlalchemy.types.Enum

Comment (anmaxvl) [reply sheepdog/blueprint/qcreport/service.py]: fyi: https://github.com/NCI-GDC/gdc-ng-models/blob/master/gdc_ng_models/models/qcreport.py#L18 + I think you're missing the idea behind this validation.

**Pull request merged**